client/scss/components/forms/_field-comment-control.scss

```diff
@@ -47,15 +44,17 @@ $button-size: theme('spacing.5');
   }
 }
 
-// Add extra space to the right of commentable fields for the buttons.
 .w-field--commentable {
+  // Allow comment buttons to be placed along side the field by display flex
   .tab-content--comments-enabled & .w-field__input {
-    padding-inline-end: $button-size;
+    display: flex;
+    align-items: center;
+  }
 
-    @include media-breakpoint-up(sm) {
-      // Partial offset to reduce the change in layout with/without comments.
-      padding-inline-end: calc($button-size / 1.5);
-    }
+  textarea ~ .w-field__comment-button,
+  &.w-field--checkbox_select_multiple .w-field__comment-button,
+  &.w-field--admin_tag_widget .w-field__comment-button {
```

Comment:
I have added this one line myself – this is an example for a widget that isn’t working with Flexbox. We’d need to additionally have a wrapping `<div>` in its widget template if we want to proceed with this approach.